### DISCUSSION TITLE
Account Linking For Multiple OAuth Accounts

### DISCUSSION THREAD

**Discussion by malek10xdev** [supabase/supabase#14564] [Feature Requests]
## Is your feature request related to a problem? Please describe.

We aim to enhance our app by enabling users to link multiple accounts from different authentication providers without requiring separate account creation.

**Issues:**

- Inability to connect multiple OAuth accounts to one user.
- Inability to at least utilize OAuth solely for obtaining provider tokens without creating an account.

## Describe the solution you'd like

Let's consider an example to better illustrate the problem.  James, a user of the Sales CRM app, is logged in using his primary Gmail account, [james@gmail.com](mailto:james@gmail.com). However, James also has another Gmail account, [james.pro@gmail.com](mailto:james.pro@gmail.com), from which he wants to import contacts into the app.

By linking his [james.pro@gmail.com](mailto:james.pro@gmail.com) account to his primary [james@gmail.com](mailto:james@gmail.com) account, James aims to import contacts from  [james.pro@gmail.com](mailto:james.pro@gmail.com) without having to create a separate account.

To achieve this desired functionality, the following capabilities are required:

1. Implementation of account linking functionality or, at minimum, the ability to only retrieve the necessary provider tokens.

## Describe alternatives you've considered

As a workaround, we have temporarily modified the Gotrue provider callback URI to redirect to our backend. This approach allows us to fetch the token and perform the necessary actions or redirect users back to Gotrue for user account creation.

## Additional context

N/A

**Comment by hf** [comment]
Planned. Please watch this issue: https://github.com/supabase/gotrue/issues/313
